src/dlls/dbgshim/dbgshim.cpp

```diff
@@ -387,25 +387,63 @@ class RuntimeStartupHelper
         return hr;
     }
 
+    bool NoNullHandles(HANDLE *handleArray, DWORD arrayLength)
+    {
+        for (int i = 0; i < (int)arrayLength; i++)
+        {
+            HANDLE h = handleArray[i];
+            if (h == NULL)
```

Comment:
Actually, it looks like there are at least some code paths where failure would be INVALID_HANDLE_VALUE (unless I am missing something)